connexion/decorators/main.py

```diff
@@ -166,10 +166,7 @@ async def wrapper(*args, **kwargs):
                 uri_parser=self.uri_parser, scope=scope, receive=receive
             )
             decorated_function = self.decorate(function)
-            response = decorated_function(request)
-            while asyncio.iscoroutine(response):
-                response = await response
-            return response
+            return await decorated_function(request)
```

Comment:
Just a fix to get the `AsyncApp` working correctly. Not related to enforcing defaults, but necessary to make the example work.